ios/FluentUI/Pill Button Bar/PillButtonBar.swift

```diff
@@ -148,6 +148,10 @@ open class PillButtonBar: UIScrollView {
         preconditionFailure("init(coder:) has not been implemented")
     }
 
+    @objc public func pillsBackgroundColor(_ backgroundColor: UIColor) {
```

Comment:
this seems strange to me. if you are going to override the entire pill buttons with a particular color, would it make more sense to do it in initialization of the pill bar? this kind of api makes more sense to me if we were providing index as well to change the pill background one by one.

Reply:
I did thought about taking it as part of init but then wasn't sure of what should go as default value for it if we put in init. hence created a separate func so that user can explicitly set it if they want.
Will re-think about setting it in init and update.

---
In reply to: [467157750](https://github.com/microsoft/fluentui-apple/pull/175#discussion_r467157750) [](ancestors = 467157750)

Reply:
could not find a good way of putting this in init as I am unsure about what to set as default if user doesn't specify. Requesting suggestions.

---
In reply to: [467421927](https://github.com/microsoft/fluentui-apple/pull/175#discussion_r467421927) [](ancestors = 467421927,467157750)

Reply:
How about this?

You remove the pillsBackgroundColor func and try the below:

`
    private var customPillButtonBackgroundColor: UIColor?

    @objc public init(pillButtonStyle: PillButtonStyle = .outline, pillButtonBackgroundColor: UIColor? = nil) {
    ...
    		self.customPillButtonBackgroundColor = pillButtonBackgroundColor
    ...
    }

    private func addButtonsWithItems(_ items: [PillButtonBarItem]) {
        for (index, item) in items.enumerated() {
            ...

            if let customButtonBackgroundColor = self.customPillButtonBackgroundColor {
                button.customBackgroundColor = customButtonBackgroundColor
            }
        }
    }
`

Reply:
Incorporated the suggestions. Thank you :)
It wasn't taking nil as default value earlier as i didn't use UIColor?

---
In reply to: [468241264](https://github.com/microsoft/fluentui-apple/pull/175#discussion_r468241264) [](ancestors = 468241264)

Reply:
Updated

---
In reply to: [467905206](https://github.com/microsoft/fluentui-apple/pull/175#discussion_r467905206) [](ancestors = 467905206,467421927,467157750)